TestNodePortLocal is very flaky on Windows

**Describe the bug**
<!--
A clear and concise description of what the bug is.
If you believe this bug is a security issue, please don't use this template and follow our [security guidelines](/SECURITY.md)
-->
```
--- FAIL: TestNodePortLocal (518.50s)
    --- FAIL: TestNodePortLocal/testNPLAddPod (279.89s)
        --- PASS: TestNodePortLocal/testNPLAddPod/NPLTestMultiplePods (68.29s)
        --- PASS: TestNodePortLocal/testNPLAddPod/NPLTestPodAddMultiPort (21.26s)
        --- FAIL: TestNodePortLocal/testNPLAddPod/NPLTestPodAddMultiProtocol (90.18s)
        --- FAIL: TestNodePortLocal/testNPLAddPod/NPLTestLocalAccess (100.17s)
    --- FAIL: TestNodePortLocal/testNPLMultiplePodsAgentRestart (98.17s)
    --- FAIL: TestNodePortLocal/testNPLChangePortRangeAgentRestart (100.16s)
```

[console.log](https://github.com/antrea-io/antrea/files/11504994/console.log)
[antrea-test-logs.tar.gz](https://github.com/antrea-io/antrea/files/11505004/antrea-test-logs.tar.gz)

--- a/build/yamls/antrea-windows-containerd.yml
+++ b/build/yamls/antrea-windows-containerd.yml
@@ -168,7 +168,7 @@ data:
     # from that range will be assigned whenever a Pod's container defines a specific port to be exposed
     # (each container can define a list of ports as pod.spec.containers[].ports), and all Node traffic
     # directed to that port will be forwarded to the Pod.
-    #  portRange: 61000-62000
+    #  portRange: 40000-41000
   antrea-cni.conflist: |
     {
         "cniVersion":"0.3.0",
@@ -187,7 +187,7 @@ kind: ConfigMap
 metadata:
   labels:
     app: antrea
-  name: antrea-windows-config-8kfkb8t957
+  name: antrea-windows-config-89b7ch9t9b
   namespace: kube-system
 ---
 apiVersion: apps/v1
@@ -271,7 +271,7 @@ spec:
         operator: Exists
       volumes:
       - configMap:
-          name: antrea-windows-config-8kfkb8t957
+          name: antrea-windows-config-89b7ch9t9b
         name: antrea-windows-config
       - configMap:
           defaultMode: 420

--- a/build/yamls/antrea-windows.yml
+++ b/build/yamls/antrea-windows.yml
@@ -158,7 +158,7 @@ data:
     # from that range will be assigned whenever a Pod's container defines a specific port to be exposed
     # (each container can define a list of ports as pod.spec.containers[].ports), and all Node traffic
     # directed to that port will be forwarded to the Pod.
-    #  portRange: 61000-62000
+    #  portRange: 40000-41000
   antrea-cni.conflist: |
     {
         "cniVersion":"0.3.0",
@@ -177,7 +177,7 @@ kind: ConfigMap
 metadata:
   labels:
     app: antrea
-  name: antrea-windows-config-cmccc6hbb4
+  name: antrea-windows-config-hc82tmf96f
   namespace: kube-system
 ---
 apiVersion: apps/v1
@@ -265,7 +265,7 @@ spec:
         operator: Exists
       volumes:
       - configMap:
-          name: antrea-windows-config-cmccc6hbb4
+          name: antrea-windows-config-hc82tmf96f
         name: antrea-windows-config
       - configMap:
           defaultMode: 420

--- a/build/yamls/windows/base/conf/antrea-agent.conf
+++ b/build/yamls/windows/base/conf/antrea-agent.conf
@@ -140,4 +140,4 @@ nodePortLocal:
 # from that range will be assigned whenever a Pod's container defines a specific port to be exposed
 # (each container can define a list of ports as pod.spec.containers[].ports), and all Node traffic
 # directed to that port will be forwarded to the Pod.
-#  portRange: 61000-62000
+#  portRange: 40000-41000

--- a/ci/jenkins/test.sh
+++ b/ci/jenkins/test.sh
@@ -411,18 +411,6 @@ function deliver_antrea_windows {
     sed -i 's/if (!(Test-Path $AntreaAgentConfigPath))/if ($true)/' hack/windows/Helper.psm1
     kubectl get nodes -o wide --no-headers=true | awk -v role="$CONTROL_PLANE_NODE_ROLE" '$3 !~ role && $1 ~ /win/ {print $1}' | while read WORKER_NAME; do
         revert_snapshot_windows ${WORKER_NAME}
-        # Some tests need us.gcr.io/k8s-artifacts-prod/e2e-test-images/agnhost:2.13 image but it is not for windows/amd64 10.0.17763
-        # Use e2eteam/agnhost:2.13 instead
-        harbor_images=("sigwindowstools-kube-proxy:v1.18.0" "agnhost:2.13" "agnhost:2.13" "agnhost:2.13" "agnhost:2.29" "agnhost:2.29" "e2eteam-jessie-dnsutils:1.0" "e2eteam-jessie-dnsutils:1.0" "e2eteam-pause:3.2" "e2eteam-pause:3.2" "e2eteam-pause:3.2" "e2eteam-busybox:1.29-windows-amd64-1809")
-        antrea_images=("sigwindowstools/kube-proxy:v1.18.0" "e2eteam/agnhost:2.13" "us.gcr.io/k8s-artifacts-prod/e2e-test-images/agnhost:2.13" "k8sprow.azurecr.io/kubernetes-e2e-test-images/agnhost:2.13" "k8s.gcr.io/e2e-test-images/agnhost:2.29" "registry.k8s.io/e2e-test-images/agnhost:2.29" "e2eteam/jessie-dnsutils:1.0" "gcr.io/kubernetes-e2e-test-images/jessie-dnsutils:1.0" "e2eteam/pause:3.2" "k8s.gcr.io/pause:3.2" "registry.k8s.io/pause:3.2" "docker.io/library/busybox:1.29")
-        common_images=("mcr.microsoft.com/windows/servercore/iis:latest")
-        # Pull necessary images in advance to avoid transient error
-        for i in "${!harbor_images[@]}"; do
-            ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "docker pull -q ${DOCKER_REGISTRY}/antrea/${harbor_images[i]} && docker tag ${DOCKER_REGISTRY}/antrea/${harbor_images[i]} ${antrea_images[i]}" || true
-        done
-        for image in "${common_images[@]}"; do
-            ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "docker pull -q ${image}" || true
-        done
 
         # Use a script to run antrea agent in windows Network Policy cases
         if [ "$TESTCASE" == "windows-networkpolicy-process" ]; then
@@ -465,6 +453,18 @@ function deliver_antrea_windows {
                 ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "docker load -i antrea-windows.tar.gz"
             fi
         fi
+        # Some tests need us.gcr.io/k8s-artifacts-prod/e2e-test-images/agnhost:2.13 image but it is not for windows/amd64 10.0.17763
+        # Use e2eteam/agnhost:2.13 instead
+        harbor_images=("sigwindowstools-kube-proxy:v1.18.0" "agnhost:2.13" "agnhost:2.13" "agnhost:2.13" "agnhost:2.29" "agnhost:2.29" "e2eteam-jessie-dnsutils:1.0" "e2eteam-jessie-dnsutils:1.0" "e2eteam-pause:3.2" "e2eteam-pause:3.2" "e2eteam-pause:3.2" "e2eteam-busybox:1.29-windows-amd64-1809")
+        antrea_images=("sigwindowstools/kube-proxy:v1.18.0" "e2eteam/agnhost:2.13" "us.gcr.io/k8s-artifacts-prod/e2e-test-images/agnhost:2.13" "k8sprow.azurecr.io/kubernetes-e2e-test-images/agnhost:2.13" "k8s.gcr.io/e2e-test-images/agnhost:2.29" "registry.k8s.io/e2e-test-images/agnhost:2.29" "e2eteam/jessie-dnsutils:1.0" "gcr.io/kubernetes-e2e-test-images/jessie-dnsutils:1.0" "e2eteam/pause:3.2" "k8s.gcr.io/pause:3.2" "registry.k8s.io/pause:3.2" "docker.io/library/busybox:1.29")
+        common_images=("mcr.microsoft.com/windows/servercore/iis:latest")
+        # Pull necessary images in advance to avoid transient error
+        for i in "${!harbor_images[@]}"; do
+            ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "docker pull -q ${DOCKER_REGISTRY}/antrea/${harbor_images[i]} && docker tag ${DOCKER_REGISTRY}/antrea/${harbor_images[i]} ${antrea_images[i]}" || true
+        done
+        for image in "${common_images[@]}"; do
+            ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "docker pull -q ${image}" || true
+        done
     done
     rm -f antrea-windows.tar.gz
 }

--- a/cmd/antrea-agent/options.go
+++ b/cmd/antrea-agent/options.go
@@ -53,7 +53,6 @@ const (
 	defaultIdleFlowExportTimeout   = "15s"
 	defaultIGMPQueryInterval       = 125 * time.Second
 	defaultStaleConnectionTimeout  = 5 * time.Minute
-	defaultNPLPortRange            = "61000-62000"
 	defaultNodeType                = config.K8sNode
 	defaultMaxEgressIPsPerNode     = 255
 )

--- a/cmd/antrea-agent/options_linux.go
+++ b/cmd/antrea-agent/options_linux.go
@@ -17,6 +17,10 @@
 
 package main
 
+const (
+	defaultNPLPortRange = "61000-62000"
+)
+
 func (o *Options) checkUnsupportedFeatures() error {
 	// All features are supported on a Linux Node.
 	return nil

--- a/cmd/antrea-agent/options_windows.go
+++ b/cmd/antrea-agent/options_windows.go
@@ -29,6 +29,10 @@ import (
 	"antrea.io/antrea/pkg/ovs/ovsconfig"
 )
 
+const (
+	defaultNPLPortRange = "40000-41000"
+)
+
 func (o *Options) checkUnsupportedFeatures() error {
 	var unsupported []string
 

--- a/pkg/agent/nodeportlocal/k8s/npl_controller.go
+++ b/pkg/agent/nodeportlocal/k8s/npl_controller.go
@@ -510,7 +510,6 @@ func (c *NPLController) handleAddUpdatePod(key string, obj interface{}) error {
 		} else {
 			nodePort = portData.NodePort
 		}
-
 		if _, ok := nplAnnotationsRequiredMap[portcache.NodePortProtoFormat(nodePort, protocol)]; !ok {
 			nplAnnotationsRequiredMap[portcache.NodePortProtoFormat(nodePort, protocol)] = types.NPLAnnotation{
 				PodPort:   port,

--- a/test/e2e/nodeportlocal_test.go
+++ b/test/e2e/nodeportlocal_test.go
@@ -36,10 +36,10 @@ import (
 )
 
 const (
-	defaultStartPort  = 61000
-	defaultEndPort    = 62000
-	updatedStartPort  = 63000
-	updatedEndPort    = 64000
+	defaultStartPort  = 26000
+	defaultEndPort    = 27000
+	updatedStartPort  = 28000
+	updatedEndPort    = 29000
 	defaultTargetPort = 80
 )
 
@@ -88,10 +88,9 @@ func TestNodePortLocal(t *testing.T) {
 	t.Run("testNPLChangePortRangeAgentRestart", func(t *testing.T) { testNPLChangePortRangeAgentRestart(t, data) })
 }
 
-func getNPLAnnotation(t *testing.T, data *TestData, r *require.Assertions, testPodName string) (string, string) {
-	var nplAnn string
+func getNPLAnnotations(t *testing.T, data *TestData, r *require.Assertions, testPodName string, conditionFn func(types.NPLAnnotation) bool) ([]types.NPLAnnotation, string) {
+	var nplAnnotations []types.NPLAnnotation
 	var testPodIP *PodIPs
-	var found bool
 
 	var err error
 	maxRetries := 0
@@ -124,7 +123,18 @@ func getNPLAnnotation(t *testing.T, data *TestData, r *require.Assertions, testP
 
 			ann := pod.GetAnnotations()
 			t.Logf("Got annotations %v for Pod with IP %v", ann, testPodIP.ipv4.String())
-			nplAnn, found = ann[types.NPLAnnotationKey]
+			nplAnn, found := ann[types.NPLAnnotationKey]
+			if !found {
+				return false, nil
+			}
+			json.Unmarshal([]byte(nplAnn), &nplAnnotations)
+			if conditionFn != nil {
+				for _, annotation := range nplAnnotations {
+					if !conditionFn(annotation) {
+						return false, nil
+					}
+				}
+			}
 			return found, nil
 		})
 		if err == nil {
@@ -134,14 +144,7 @@ func getNPLAnnotation(t *testing.T, data *TestData, r *require.Assertions, testP
 		time.Sleep(time.Millisecond * 100)
 	}
 	r.NoError(err, "Poll for Pod check failed")
-	return nplAnn, testPodIP.ipv4.String()
-}
-
-func getNPLAnnotations(t *testing.T, data *TestData, r *require.Assertions, testPodName string) ([]types.NPLAnnotation, string) {
-	nplAnnotationString, testPodIP := getNPLAnnotation(t, data, r, testPodName)
-	var nplAnnotations []types.NPLAnnotation
-	json.Unmarshal([]byte(nplAnnotationString), &nplAnnotations)
-	return nplAnnotations, testPodIP
+	return nplAnnotations, testPodIP.ipv4.String()
 }
 
 func checkNPLRules(t *testing.T, data *TestData, r *require.Assertions, nplAnnotations []types.NPLAnnotation, antreaPod, podIP string, nodeName string, present bool) {
@@ -343,6 +346,19 @@ func checkTrafficForNPL(data *TestData, r *require.Assertions, nplAnnotations []
 	}
 }
 
+func getTwoNodes() (string, string) {
+	clientNode := nodeName(0)
+	serverNode := nodeName(1)
+
+	if len(clusterInfo.windowsNodes) > 1 {
+		// Same test topology on Windows and Linux testbeds.
+		clientNode = workerNodeName(clusterInfo.windowsNodes[0])
+		serverNode = workerNodeName(clusterInfo.windowsNodes[1])
+	}
+
+	return clientNode, serverNode
+}
+
 func testNPLAddPod(t *testing.T, data *TestData) {
 	t.Run("NPLTestMultiplePods", func(t *testing.T) { NPLTestMultiplePods(t, data) })
 	t.Run("NPLTestPodAddMultiPort", func(t *testing.T) { NPLTestPodAddMultiPort(t, data) })
@@ -362,12 +378,7 @@ func NPLTestMultiplePods(t *testing.T, data *TestData) {
 	annotation := make(map[string]string)
 	annotation[types.NPLEnabledAnnotationKey] = "true"
 	ipFamily := corev1.IPv4Protocol
-	clientNode := nodeName(0)
-	serverNode := nodeName(1)
-	if len(clusterInfo.windowsNodes) != 0 {
-		serverNode = workerNodeName(clusterInfo.windowsNodes[0])
-		clientNode = workerNodeName(clusterInfo.windowsNodes[1])
-	}
+	clientNode, serverNode := getTwoNodes()
 	testData.createNginxClusterIPServiceWithAnnotations(serverNode, false, &ipFamily, annotation)
 	var testPods []string
 
@@ -390,7 +401,7 @@ func NPLTestMultiplePods(t *testing.T, data *TestData) {
 
 	expectedAnnotations := newExpectedNPLAnnotations(defaultStartPort, defaultEndPort).Add(nil, defaultTargetPort, "tcp")
 	for _, testPodName := range testPods {
-		nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName)
+		nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName, nil)
 
 		checkNPLRules(t, testData, r, nplAnnotations, antreaPod, testPodIP, serverNode, true)
 		expectedAnnotations.Check(t, nplAnnotations)
@@ -406,11 +417,7 @@ func NPLTestMultiplePods(t *testing.T, data *TestData) {
 func NPLTestPodAddMultiPort(t *testing.T, data *TestData) {
 	r := require.New(t)
 
-	clientNode := nodeName(0)
-	serverNode := nodeName(1)
-	if len(clusterInfo.windowsNodes) != 0 {
-		serverNode = workerNodeName(clusterInfo.windowsNodes[0])
-	}
+	clientNode, serverNode := getTwoNodes()
 	testPodName := randName("test-pod-")
 
 	annotation := make(map[string]string)
@@ -446,7 +453,7 @@ func NPLTestPodAddMultiPort(t *testing.T, data *TestData) {
 	}).Create(data)
 	r.NoError(err, "Error creating test Pod: %v", err)
 
-	nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName)
+	nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName, nil)
 
 	clientName := randName("test-client-")
 	err = testData.createAgnhostPodOnNode(clientName, data.testNamespace, clientNode, false)
@@ -473,12 +480,7 @@ func NPLTestPodAddMultiPort(t *testing.T, data *TestData) {
 func NPLTestPodAddMultiProtocol(t *testing.T, data *TestData) {
 	r := require.New(t)
 
-	clientNode := nodeName(0)
-	serverNode := nodeName(1)
-	if len(clusterInfo.windowsNodes) != 0 {
-		serverNode = workerNodeName(clusterInfo.windowsNodes[0])
-		clientNode = workerNodeName(clusterInfo.windowsNodes[1])
-	}
+	clientNode, serverNode := getTwoNodes()
 	testPodName := randName("test-pod-")
 
 	annotation := make(map[string]string)
@@ -501,7 +503,7 @@ func NPLTestPodAddMultiProtocol(t *testing.T, data *TestData) {
 	err := NewPodBuilder(testPodName, data.testNamespace, agnhostImage).OnNode(serverNode).WithContainerName(containerName).WithCommand(cmd).WithArgs(args).WithPorts([]corev1.ContainerPort{port}).WithLabels(selector).Create(testData)
 	r.NoError(err, "Error creating test Pod: %v", err)
 
-	nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName)
+	nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName, nil)
 
 	clientName := randName("test-client-")
 	err = testData.createAgnhostPodOnNode(clientName, data.testNamespace, clientNode, false)
@@ -534,12 +536,7 @@ func NPLTestLocalAccess(t *testing.T, data *TestData) {
 	annotation[types.NPLEnabledAnnotationKey] = "true"
 	ipFamily := corev1.IPv4Protocol
 
-	clientNode := nodeName(0)
-	serverNode := nodeName(1)
-	if len(clusterInfo.windowsNodes) != 0 {
-		serverNode = workerNodeName(clusterInfo.windowsNodes[0])
-		clientNode = workerNodeName(clusterInfo.windowsNodes[1])
-	}
+	clientNode, serverNode := getTwoNodes()
 
 	testData.createNginxClusterIPServiceWithAnnotations(serverNode, false, &ipFamily, annotation)
 	expectedAnnotations := newExpectedNPLAnnotations(defaultStartPort, defaultEndPort).Add(nil, defaultTargetPort, "tcp")
@@ -558,7 +555,7 @@ func NPLTestLocalAccess(t *testing.T, data *TestData) {
 	antreaPod, err := testData.getAntreaPodOnNode(serverNode)
 	r.NoError(err, "Error when getting Antrea Agent Pod on Node '%s'", serverNode)
 
-	nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName)
+	nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName, nil)
 
 	checkNPLRules(t, testData, r, nplAnnotations, antreaPod, testPodIP, serverNode, true)
 	expectedAnnotations.Check(t, nplAnnotations)
@@ -581,14 +578,7 @@ func testNPLMultiplePodsAgentRestart(t *testing.T, data *TestData) {
 	annotation[types.NPLEnabledAnnotationKey] = "true"
 	ipFamily := corev1.IPv4Protocol
 
-	clientNode := nodeName(0)
-	serverNode := nodeName(1)
-	winenv := false
-	if len(clusterInfo.windowsNodes) != 0 {
-		serverNode = workerNodeName(clusterInfo.windowsNodes[0])
-		clientNode = workerNodeName(clusterInfo.windowsNodes[1])
-		winenv = true
-	}
+	clientNode, serverNode := getTwoNodes()
 	data.createNginxClusterIPServiceWithAnnotations(serverNode, false, &ipFamily, annotation)
 	var testPods []string
 	var err error
@@ -610,7 +600,7 @@ func testNPLMultiplePodsAgentRestart(t *testing.T, data *TestData) {
 	r.NoError(err, "Error when getting Antrea Agent Pod on Node '%s'", serverNode)
 
 	// Delete one iptables rule to ensure it gets re-installed correctly on restart.
-	nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPods[0])
+	nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPods[0], nil)
 	r.Len(nplAnnotations, 1)
 	// Make sure the rule is present first. It should always be the case if the Pod was already
 	// annotated.
@@ -623,10 +613,10 @@ func testNPLMultiplePodsAgentRestart(t *testing.T, data *TestData) {
 		podPort:  nplAnnotations[0].PodPort,
 		protocol: protocolToString(corev1.ProtocolTCP),
 	}
-	if !winenv {
-		deleteNPLRuleFromIPTables(t, data, r, antreaPod, ruleToDelete)
-	} else {
+	if len(clusterInfo.windowsNodes) > 1 {
 		deleteNPLRuleFromNetNat(t, data, r, antreaPod, ruleToDelete)
+	} else {
+		deleteNPLRuleFromIPTables(t, data, r, antreaPod, ruleToDelete)
 	}
 
 	err = data.RestartAntreaAgentPods(defaultTimeout)
@@ -637,7 +627,7 @@ func testNPLMultiplePodsAgentRestart(t *testing.T, data *TestData) {
 
 	expectedAnnotations := newExpectedNPLAnnotations(defaultStartPort, defaultEndPort).Add(nil, defaultTargetPort, "tcp")
 	for _, testPodName := range testPods {
-		nplAnnotations, testPodIP := getNPLAnnotations(t, data, r, testPodName)
+		nplAnnotations, testPodIP := getNPLAnnotations(t, data, r, testPodName, nil)
 
 		checkNPLRules(t, testData, r, nplAnnotations, antreaPod, testPodIP, serverNode, true)
 		expectedAnnotations.Check(t, nplAnnotations)
@@ -659,12 +649,7 @@ func testNPLChangePortRangeAgentRestart(t *testing.T, data *TestData) {
 	annotation[types.NPLEnabledAnnotationKey] = "true"
 	ipFamily := corev1.IPv4Protocol
 
-	clientNode := nodeName(0)
-	serverNode := nodeName(1)
-	if len(clusterInfo.windowsNodes) != 0 {
-		serverNode = workerNodeName(clusterInfo.windowsNodes[0])
-		clientNode = workerNodeName(clusterInfo.windowsNodes[1])
-	}
+	clientNode, serverNode := getTwoNodes()
 	data.createNginxClusterIPServiceWithAnnotations(serverNode, false, &ipFamily, annotation)
 	var testPods []string
 	var err error
@@ -684,7 +669,7 @@ func testNPLChangePortRangeAgentRestart(t *testing.T, data *TestData) {
 
 	var rules []nplRuleData
 	for _, testPodName := range testPods {
-		nplAnnotations, testPodIP := getNPLAnnotations(t, data, r, testPodName)
+		nplAnnotations, testPodIP := getNPLAnnotations(t, data, r, testPodName, nil)
 		for i := range nplAnnotations {
 			for j := range nplAnnotations[i].Protocols {
 				rule := nplRuleData{
@@ -697,7 +682,6 @@ func testNPLChangePortRangeAgentRestart(t *testing.T, data *TestData) {
 			}
 		}
 	}
-
 	configureNPLForAgent(t, data, updatedStartPort, updatedEndPort)
 
 	antreaPod, err := data.getAntreaPodOnNode(serverNode)
@@ -714,7 +698,10 @@ func testNPLChangePortRangeAgentRestart(t *testing.T, data *TestData) {
 	}
 
 	for _, testPodName := range testPods {
-		nplAnnotations, testPodIP := getNPLAnnotations(t, data, r, testPodName)
+		conditionFn := func(ann types.NPLAnnotation) bool {
+			return ann.NodePort >= updatedStartPort
+		}
+		nplAnnotations, testPodIP := getNPLAnnotations(t, data, r, testPodName, conditionFn)
 
 		checkNPLRules(t, data, r, nplAnnotations, antreaPod, testPodIP, serverNode, true)
 		expectedAnnotations.Check(t, nplAnnotations)

